lecture_3/template/nkt2111.tex

```diff
@@ -0,0 +1,255 @@
+    \item we walked through the homework
+    \begin{itemize}
+        \item for number 1, we do not need to write code but we do need to be detailed and think about what resources you need for each problem
+        \item when a problem notes changes in the size of the data set, we should try to find the practical solution for each size rather than just using the fanciest methods for each 
+    \end{itemize}
+\end{itemize}
+
+Next, we moved on to a review of the command line using citi bike data. Here are the commands that we explored:
+\begin{itemize}
+    \item wc -l 201402-citibike-tripdata.csv \\
+    Counts the number of lines in the data set
+    \item -d, -f15 201402-citibike-tripdata.csv \textbar head \\
+    Extracts rider gender in column 15, specifying ',' as a delimiter and limits output to first 10 lines
+    \item cut -d, -f14 201402-citibike-tripdata.csv \textbar sort \textbar head \\
+    Find the earliest birth year in column 14. Exchanging "tail" for "head" would get us the latest birth year
+    \item grep Broadway 201402-citibike-tripdata.csv \textbar head \\
+    Finds all trips either starting or ending on Broadway
+    \item cut -d, -f5,9 201402-citibike-tripdata.csv \textbar grep 'Broadway.*Broadway' \textbar wc -l \\
+    List all of the unique stations in column 5 that are contain Broadway by first sorting and then removing running duplicates
+    \item cut -d, -f14 201402-citibike-tripdata.csv \textbar grep '[0-9]' \textbar sort \textbar tail \\
+    find the latest birth year in column 14, limiting to lines with a number. Here '[0-9]' means anything between 0 and 9
+    \item cut -d, -f15 201402-citibike-tripdata.csv \textbar sort \textbar uniq -c \\
+    Counts trips by gender. \\
+    Interesting note: here, we found that more than 4 times the riders are male than female.
+    \item We noted some particular subcommands individually. "-r" can be used to reverse sort. "-k" denotes a key, "wc" tells the number of characters, lines, and words, quotes are used in pattern matching
+\end{itemize}
+
+Next, we were briefly introduced to awk, which is extremely powerful.
+\begin{itemize}
+    \item awk -F, '\$5 ~ /Broadway/ \&\& \$9 ~ /Broadway/' 201402-citibike-tripdata.csv \textbar wc -l \\
+    use awk to count all trips that start and end on Broadway
+    \item awk -F, '{counts[\$15]++} END {for (k in counts) print counts[k]"\t" k }' 201402-citibike-tripdata.csv \\
+    use awk to count trips by gender without having to sort. It is interesting to note here the function of {counts[\$15]++}. This makes an array, without any initialization, and counts the number of appearances of each item in column 15
+    \item cut -d, -f14 201402-citibike-tripdata.csv \textbar sort \textbar uniq -c \textbar awk '{printf \$2"\t"; for (i=1; i<=\$1/100; i++) printf "*"; printf "\n"}' \\
```

Comment:
something isn't escaped properly here which is the last thing breaking the build.

one way to get around these problems is to use either the listings package or the verbatim package so that you can drop code straight into the document without having to escape everything. so either catch the missed escape or try one of those.